vespamalloc/src/vespamalloc/malloc/mmappool.cpp

```diff
@@ -76,9 +83,10 @@ MMapPool::mmap(size_t sz) {
         std::lock_guard guard(_mutex);
         auto [it, inserted] = _mappings.insert(std::make_pair(buf, MMapInfo(mmapId, sz)));
         ASSERT_STACKTRACE(inserted);
+        size_t sum(0);
+        std::for_each(_mappings.begin(), _mappings.end(), [&sum](const auto & e){ sum += e.second._sz; });
```

Comment:
Since this is tracking the peak usage, could we instead increase a "current usage" sum on mmap and decrease it on unmap, then maintain the max value observed of this? The O(n) CPU time per mapping seems like it could add up over time...